src/include/cell.i

```diff
@@ -256,7 +281,9 @@ __wt_cell_pack_addr(WT_SESSION_IMPL *session,
 {
 	uint8_t *p;
 
-	p = cell->__chunk + 1;
+	cell->__chunk[0] = '\0';
+	p = cell->__chunk;
```

Comment:
Maybe worth a comment: it isn't immediately clear to me why we were skipping a byte here before, and now we have to initialize it here.

Reply:
This is messy.

Previous versions of these functions set `p` to point past the cell descriptor byte, then used `p` to store data and other values that are associated with the cell descriptor byte, and somewhere in the top-level function did an explicit assignment of some value to `cell->__chunk[0]`. (That meant we didn't need to clear `cell->__chunk[0]`).

Current versions call underlying timestamp functions to optionally store timestamp information, in some cases setting a bit in the cell descriptor byte. That means we need to set `p` to the cell descriptor byte so those functions can set that bit, and we need to clear the cell descriptor byte because the function might not set the bit. When the underlying timestamp function returns, we set additional bits in the cell descriptor byte (so, instead of `cell->__chunk[0] = XXX`, it's `cell->__chunk[0] |= XXX`).

I tried a couple of different approaches, for example, clearing the byte in the underlying timestamp function. Anyway, everything I tried made the various cell functions diverge and look different -- value cells may or may not store timestamps, address cells always store timestamps, and key cells don't have timestamps at all.

So, I ended up with:
```
cell->__chunk[0] = '\0';
p = cell->__chunk[0];
```
as a function prefix everywhere, that is, always clear the cell descriptor byte and set `p` to point to it, even in cases where it's not necessary (for example, because address cells don't set a timestamp bit, we could pass the function that writes address timestamps a pointer past the cell descriptor byte). I thought optimizing those cases was unlikely to be useful (and the compiler is likely to catch those cases anyway), so I just gave every function the same prefix, including clearing the cell descriptor byte.

There's also an obvious question if there's a better overall structure: could we layout the cell without jumping around so much?

tl;dr If you have a suggestion of a better way to structure this code, or a suggestion about what that comment might look like, feel free to add it, or I can.

Reply:
In a055e22, I turned this into:
```
	/* Start building a cell: the descriptor byte starts zero. */
	p = cell->__chunk;
	*p = '\0';
```
That said, this was mostly for my benefit understanding the change: if you prefer it the other way, I won't fight.